Makefile

```diff
@@ -0,0 +1,59 @@
+ENV_TYPE ?= venv
+PYTHON_VERSION ?= 3.10
+VENV_DIR := .venv
```

Comment:
This comment might be too late now, but with `:=` the variable is immediately assigned the value at the time it is declared, and it cannot be overridden later when invoking make, whereas `?=` the value is only assigned if the var is not already defined (either in the Makefile itself or as a command-line override when invoking make).

Reply:
Ok, that's what I thought from looking it up, I was just confused because when I called `make list-args VENV_DIR=dummy` it prints the overridden value